guides/security.md

```diff
@@ -0,0 +1,524 @@
+
+  <div>
+    <button type="submit">Save</button>
+  </div>
+</form>
+
+```
+
+This maps to the following HTTP request:
+
+```text
+POST /posts HTTP/1.1
+Host: example.com
+Content-Type: application/x-www-form-urlencoded
+Content-Length: 53
+
+post[title]=My+Title&post[body]=This+is+the+body
+```
+
+An attacker can embed the following form on attacker.com:
+
+```html
+<form action="https://example.com/posts" method="post">
+
+  <label for="post_title">Title</label>
+  <input id="post_title" name="post[title]" type="text" value="Attacker post">
+
+  <label for="post_body">Body</label>
+  <textarea id="post_body" name="post[body]">This post was written by the bad guy</textarea>
+
+  <div>
+    <button type="submit">Save</button>
+  </div>
+</form>
+```
+
+Note that this form does not even have to be visible to the victim user. When the user visits `attacker.com` the form will automatically submit a POST request on behalf of the victim, with the victim's current session cookie, to the vulnerable site. 
+
+The way most web frameworks, including Phoenix, mitigate this vulnerability is by requiring a CSRF token when submitting a form.
+
+```text
+# A typical CSRF token seen in a Phoenix form
+<input name="_csrf_token" type="hidden" hidden="" 
+  value="WUZXJh07BhAIJ24jP1d-KQEpLwYmMDwQ0-2eYNLH_x8oHoO_qv_HJDqZ">
+```
+
+This changes the previous HTTP request to:
+
+```text
+POST /posts HTTP/1.1
+Host: example.com
+Content-Type: application/x-www-form-urlencoded
+Content-Length: 53
+
+post[title]=My+Title&post[body]=This+is+the+body&post[_csrf_token]=WUZXJh07BhAIJ24jP1d-KQEpLwYmMDwQ0-2eYNLH_x8oHoO_qv_HJDqZ
+```
+
+Because the token is randomly generated, an attacker cannot predict what the value will be. The application is safe against CSRF attacks because Phoenix checks the value of this token by default via the `:protect_from_forgery` plug, which is included in the default `:browser` pipeline of a new Phoenix project:
+
+```
+# router.ex
+
+  pipeline :browser do
+    plug :accepts, ["html"]
+    plug :fetch_session
+    plug :fetch_live_flash
+    plug :put_root_layout, {CarafeWeb.LayoutView, :root}
+    plug :protect_from_forgery      # <-- Checks the CSRF token value
+    plug :put_secure_browser_headers
+    plug :fetch_current_user
+  end
+```
+
+CSRF protections are included in Phoenix by default, so your application is most likely not vulnerable if you are using the default settings. 
+
+### Action Re-Use CSRF 
+
+Most descriptions of CSRF focus on state changing POST requests and the need for random tokens, as was just covered. Is CSRF possible with a GET request? Yes, GET requests should never be used to perform state changing actions in a web application (place an order, transfer money, etc) because they cannot be protected against CSRF in the same way POST requests can. 
+
+Consider the following form:
+
+```text
+<.form let={f} for={@bio_changeset} action={Routes.user_settings_path(@conn, :edit_bio)} method={"post"} id="edit_bio">
```

Comment:
`:let` needs the colon, and maybe we can also use the newer verified routes in the example?

```suggestion
```heex
<.form :let={f} for={@bio_changeset} action={~p"/users/settings/edit_bio"} method="post" id="edit_bio">
```